src/monaca.js

```diff
@@ -2027,6 +2027,25 @@
     argvs = argvs || [];
     var deferred = Q.defer();
 
+    if (!npm) {
+      var npmModules = ['npm', 'global-npm'];
+
+      for (var i in npmModules) {
+        if (npm) {
```

Comment:
I think finding the appropriate `npm` module should be separated in another private function. Also, I think `return` in line 2035 can be written after the require is successful. Is there any intention you put this at the beginning of the `for` loop?